
diff --git a/src/amitk_roi.h b/src/amitk_roi.h
index 2b5f355..a6051e9 100644
--- a/src/amitk_roi.h
+++ b/src/amitk_roi.h
@@ -152,7 +152,7 @@ void            amitk_roi_calculate_on_data_set   (const AmitkRoi * roi,
 						   const guint gate,
 						   const gboolean inverse,
 						   const gboolean accurate,
-						   void (* calculation)(),
+						   void (* calculation)(AmitkVoxel, amide_data_t, amide_real_t, gpointer),
 						   gpointer data);
 void            amitk_roi_erase_volume            (const AmitkRoi * roi, 
 						   AmitkDataSet * ds,

