src/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Makefile.am b/src/Makefile.am
index 8d544e1..e64e5d5 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,3 +1,4 @@
+.NOTPARALLEL:
 if AMIDE_OS_WIN32
 
 AMIDE_LDADD_WIN32 = -lwsock32 ../win32/amiderc.o

Changes:

src/Makefile.am
  @@ -1,3 +1,4 @@
  +.NOTPARALLEL:
   if AMIDE_OS_WIN32
   
   AMIDE_LDADD_WIN32 = -lwsock32 ../win32/amiderc.o
  +1 -0
